### DISCUSSION TITLE
x-data maximum string length

### DISCUSSION THREAD

**Discussion by ItalicIntegral** [alpinejs/alpine#3400] [7. Bugs (archived)]
I am unable to put in more than a 4000 character long json string in my x-data field. Is this an Alpine.js limitation? I was unable to find anything in the documentation on this.

I tried revising my code to register a store and using that. Now I can only get over 2000 characters or so.

**Comment by ekwoka** [comment]
If you're putting that much in an x-data field, it should not be in the x-data field at all.

You should use `Alpine.data`

There isn't any reason why it shouldn't work though. 

Also, in the future, please provide details of the actual errors encountered, etc. and example code.

"doesn't work" is not useful debugging information.
